DOCS-781: Hide secret in sample code

andf-viam opened the pull request:
Add additional step to guidance involving the Code sample tab: toggle **Include secret** before copy&paste.

andf-viam commented on docs/program/sdks/_index.md:
````diff
@@ -157,52 +157,31 @@ The TypeScript SDK currently only supports building web browser apps.
+main().catch((error) => {
+  console.error('encountered an error:', error)
+});
 ```
````
Looks like we have largely changed this code sample since initial publish. This is the current code sample content shown today.

sguequierre reviewed (approved):
Awesome but I would say to make this more specific by saying "robot location secret". Instead of "robot secret", with robot address and robot part secret we have been leaving location secret specifics murky across docs

andf-viam commented on docs/program/sdks/_index.md:
```diff
@@ -227,9 +206,9 @@ The C++ SDK is currently in alpha.
+  std::string host("rover-main-main.yriemen2i2.viam.cloud");
   DialOptions dial_opts;
-  Credentials credentials("SECRET FROM THE VIAM APP");
+  Credentials credentials("d7fxn4zc92cjwmlhd7zh545la0xmpc71nnn9gmtyb28g0nhd");
```
Since we're instructing users to show the secret now, this will appear populated. Elsewhere as well.

npentrel replied:
If you want to show something show a credential that is recognizably fake like `a1bcd2ef34ghijklmn5op678qr9stuvw01xyz2abcde12f3ghi` but I think it's easier for readers to see what they need to change if we keep the instructions.

You can also move the instruction to get the secret and address to the next step after the code and add some instruction to store the credentials safely.

andf-viam commented:
Sorry for all the builds (and maybe notifications? ugh, sorry!). I cleaned up a lot and made everything more consistent. If possible, I'd love a follow-up review to make sure I got it all! Thanks & no rush ofc!

npentrel commented on docs/program/sdks/_index.md:
```diff
@@ -95,12 +95,12 @@ from viam.rpc.dial import Credentials, DialOptions
         type='robot-location-secret',
-        payload='SECRET FROM THE VIAM APP')
+        payload='d7fxn4zc92cjwmlhd7zh545la0xmpc71nnn9gmtyb28g0nhd')
     opts = RobotClient.Options(
         refresh_interval=0,
         dial_options=DialOptions(credentials=creds)
     )
-    return await RobotClient.at_address('ADDRESS FROM THE VIAM APP', opts)
+    return await RobotClient.at_address('rover-main-main.yriemen2i2.viam.cloud', opts)
```
guessing this isn't on purpose?

npentrel commented on docs/program/sdks/_index.md:
```diff
@@ -133,11 +133,11 @@ func main() {
       context.Background(),
-      "ADDRESS FROM THE VIAM APP",
+      "rover-main-main.yriemen2i2.viam.cloud",
       logger,
       client.WithDialOptions(rpc.WithCredentials(rpc.Credentials{
           Type:    utils.CredentialsTypeRobotLocationSecret,
-          Payload: "SECRET FROM THE VIAM APP",
+          Payload: "d7fxn4zc92cjwmlhd7zh545la0xmpc71nnn9gmtyb28g0nhd",
```
is this on purpose?

npentrel commented on docs/program/sdks/_index.md:
```diff
@@ -157,52 +157,31 @@ The TypeScript SDK currently only supports building web browser apps.
     host,
-    credential,
+    credential: {
+      type: 'robot-location-secret',
+      payload: 'd7fxn4zc92cjwmlhd7zh545la0xmpc71nnn9gmtyb28g0nhd',
```
remove secrets

npentrel reviewed (commented):
Don't worry about notifications, I think we all tend to have a lot of builds on a PR :). 

I'm not sure whether showing a secret and address in the code snippets is a good idea. People may not understand that those aren't their credentials. They may think the app populates them into the docs for them or something like that. Willing to be convinced but I think at minimum we need to add a comment in the code to that effect

andf-viam commented:
Thanks for the feedback all! Ready for another pass: I removed all secrets and robot addresses, and added 'get your own values from the Code sample tab" additional step as applicable. LMK what you think!

npentrel commented on docs/components/arm/_index.md:
```diff
@@ -61,7 +61,8 @@ Supported arm models include:
 
-To get started using Viam's SDKs to connect to and control your robot, go to your robot's page on [the Viam app](https://app.viam.com), navigate to the **Code Sample** tab, select your preferred programming language, and copy the sample code generated.
+To get started using Viam's SDKs to connect to and control your robot, go to your robot's page on [the Viam app](https://app.viam.com) and navigate to the **Code sample** tab.
+Then, select your preferred programming language, toggle **Include secret** to display your robot's location secret in the code, and copy the sample code generated.
```
Sorry - I think we can still improve this further. I don't really like the flow we are suggesting to the user because following it makes it easy for users to introduce security risks in their code if they push it to GH. It may be worth turning this into an include because we will probably change it further as the UI gets changed. How about something like this:

```suggestion
Then, select your preferred programming language, and copy the sample code generated.

By default, the copied sample code does not include your **location secret**.
We strongly recommend adding your location secret as an environment variable and importing is as needed.

You can obtain the [location secret]() by toggleing **Include secret** to display your robot's location secret in the code or on the [Location's page](/manage/fleet/locations/#share-a-location-using-location-secret-keys).

{{% snippet "secret-share.md" %}}
```

npentrel reviewed (changes requested):
Sorry - one more concern

andf-viam commented on static/include/snippet/show-secret.md:
```diff
@@ -0,0 +1,9 @@
+You can also see your location secret on the [locations page](/manage/fleet/locations/).
+{{< /alert >}}
+
+{{% snippet "secret-share.md" %}}
```
Nesting snippets because:

- There are some places where we only want to warn of the dangers of sharing, but
- Everywhere else we want to do both:
	- warn of dangers of sharing and also:
	- advise on how to find and show location secrets.

So I include the other snippet in this one: meaning we always get the warning when we show users how to access their secret, but are still able to provide just the warning via `{{% snippet "secret-share.md" %}}` directly as needed (which we do in a few places, like when talking about parts secrets or when not discussing the **Code sample** tab).

andf-viam commented:
Ok, ready for another look. Converted to a snippet and reused extensively. Many places need both the "how to show" and "careful with secrets!" snippets, so I'm nesting the latter into the former, so we can still use it on its own in a few places. Here's a [familiar example](https://docs-test.viam.dev/67e133ce9e24b18a6c1c20d817e311aceb58ff48/public/tutorials/projects/guardian/#connect) with the snippets+alerts approach.

 LMK what you think, or if you might like to see this without alert formatting, or what not. Thanks!

npentrel commented on docs/components/movement-sensor/_index.md:
```diff
@@ -42,8 +42,9 @@ Model | Description <a name="model-table"></a>
-Then, select your preferred programming language, toggle **Include secret** to display your robot's location secret in the code, and copy the sample code generated.
+To get started using Viam's SDKs to connect to and control your robot, go to your robot's page on [the Viam app](https://app.viam.com), navigate to the **Code sample** tab, select your preferred programming language, and copy the sample code generated.
+
+{{% snippet "show-secret.md" %}}
```
this is fine - but worth pointing out that you could've also included the previous line in the snippet

npentrel commented on docs/manage/configuration.md:
````diff
@@ -21,7 +21,9 @@ The credentials look like this:
 ```
 
-Your robot's location secret can be found on the **Code sample** tab on the Viam app. Toggle **Include secret** to display it in the code sample shown there.
+Your robot's location secret can be found on the **Code sample** tab on the Viam app.
````
```suggestion
Your robot's location secret can be found on the **Code sample** tab on the Viam app or on the location's page.
```

npentrel commented on docs/tutorials/services/accessing-and-moving-robot-arm.md:
````diff
@@ -371,7 +373,9 @@ if __name__ == '__main__':
 ```
 
-Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com). Toggle **Include secret** to display these values in the code sample shown there.
+Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com).
````
```suggestion
Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com) or on the Location's page.
```

npentrel commented on docs/tutorials/services/plan-motion-with-arm-gripper.md:
````diff
@@ -472,7 +472,9 @@ if __name__ == '__main__':
 ```
 
-Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com). Toggle **Include secret** to display these values in the code sample shown there.
+Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com).
````
```suggestion
Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com) or on the {{< glossary_tooltip term_id="location" text="location" >}}'s page.
```

npentrel commented on static/include/snippet/show-secret.md:
```diff
@@ -0,0 +1,9 @@
+We strongly recommend that you add your location secret as an environment variable and import this variable into your development environment as needed.
+
+To show your robot's location secret in the sample code, toggle **Include secret** on the **Code sample** tab.
+You can also see your location secret on the [locations page](/manage/fleet/locations/).
```
This may not work but worth trying
```suggestion
You can also see your location secret on the {{< glossary_tooltip term_id="location" text="location" >}}'s page.
```

andf-viam replied:
Hmm, not sure I see the value here: the user is not trying to better understand our fleet approach here, but trying to view and copy their secret. That secret lives on the locations page in the app, which I am trying to help them find by linking to that page's "reference page" in our docs (`/fleet/manage/`). While the tooltip clarification is helpful in general, I don't think it answers the most likely question here: "where can I get my secret?", which the linked-to page does attempt to do.

Happy to hear your reasoning!

npentrel commented on docs/tutorials/projects/guardian.md:
````diff
@@ -469,7 +470,9 @@ async def connect():
 ```
 
-Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com). Toggle **Include secret** to display these values in the code sample shown there.
+Your robot's location secret and address can be found on the **Code sample** tab on [the Viam app](https://app.viam.com).
````
This info is in show-secret
```suggestion
```

npentrel reviewed (approved):
LGTM once you've considered these final comments

npentrel commented on docs/tutorials/projects/integrating-viam-with-openai.md:
```diff
@@ -147,7 +147,9 @@ pip install -r requirements.txt
 
 Finally, you will need both Viam robot credentials and OpenAI API credentials in order to run the software.
-Viam credentials can be copied from the **Code sample** tab on your [Viam robot page](https://app.viam.com). Click **Include secret** to unhide the credentials.
+Viam credentials can be copied from the **Code sample** tab on your [Viam robot page](https://app.viam.com).
```
I think you missed this one @andf-viam otherwise good to merge
